org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/files/generateAdaptorApplication.mtl

```diff
@@ -81,11 +79,14 @@ import org.eclipse.lyo.oslc4j.core.model.Property;
 import org.eclipse.lyo.oslc4j.core.model.Publisher;
 import org.eclipse.lyo.oslc4j.core.model.QueryCapability;
 import org.eclipse.lyo.oslc4j.core.model.ResourceShape;
+import org.eclipse.lyo.oslc4j.core.model.ResourceShapeFactory;
 import org.eclipse.lyo.oslc4j.core.model.Service;
 import org.eclipse.lyo.oslc4j.core.model.ServiceProvider;
 import org.eclipse.lyo.oslc4j.core.model.ServiceProviderCatalog;
 import org.eclipse.lyo.oslc4j.provider.jena.JenaProvidersRegistry;
 import org.eclipse.lyo.oslc4j.provider.json4j.Json4JProvidersRegistry;
+//TODO: this is dependent on Jersey. How to produce code that is independent of the jax-rs implementation?
```

Comment:
This is a major issue. 
Is it even possible to generate an adaptor without caring about the jax-rs implementation?

Reply:
I hope I found the solution, but have not tested it yet.
I can extend my Application class from javax.ws.rs.core.Application, instead of ResourceConfig.
The method getClasses() seems to be the same as registerClasses() from Jersey.
Section 29.3.7.1 from https://docs.oracle.com/javaee/7/tutorial/jaxrs002.htm

Only problem is that I already call the class Application. (I do recall @jamsden warning about this time some time ago).

Reply:
There is no problem with the naming really:

`public class Application extends javax.ws.rs.core.Application`

I would suggest to call it `LyoApplication` however.

I have just changed my code to extend the `Application` directly and the rest of the code looks like this now:

```java
    public Application()
           throws OslcCoreApplicationException,
                  URISyntaxException
    {
        final String BASE_URI = "http://localhost/validatingResourceShapes";
        for (final Map.Entry<String, Class<?>> entry : RESOURCE_SHAPE_PATH_TO_RESOURCE_CLASS_MAP.entrySet()) {
            ResourceShapeFactory.createResourceShape(BASE_URI, OslcConstants.PATH_RESOURCE_SHAPES,
                entry.getKey(), entry.getValue());
        }
    }

    @Override
    public Set<Class<?>> getClasses() {
        return RESOURCE_CLASSES;
    }
```

I think it is rather silly to  do the initialisation in the static ctor. If the new base class works, we can swap the initialisation procedure too.

Reply:
And bingo, it works like a charm:

```
scott_sandbox-twin.1.xy7al343d0ni@linuxkit-025000000001    | 1026 [INFO] Application - Initialising the LyoApplication
scott_sandbox-twin.1.xy7al343d0ni@linuxkit-025000000001    | 1704 [TRACE] Application - Application.getClasses() was called
scott_sandbox-twin.1.xy7al343d0ni@linuxkit-025000000001    | 1729 [TRACE] Application - Application.getClasses() was called
```

Reply:
1. You did not have to add "extends javax.ws.rs.core.Application" to our Application class?

2. 
>I think it is rather silly to do the initialisation in the static ctor

What do you mean? please clarify, or suggest code change.

Reply:
chatted with Jad, resolved now. @neormx are you interested in this for your work? Would be good if you could verify it works for you. I recall you use WildFly instead of Jersey? No need to check it before it is merged, you check if the newly generated code works for you.